src/pygama/math/utils.py

```diff
@@ -102,4 +104,76 @@ def print_fit_results(pars: np.ndarray, cov: np.ndarray, func: Optional[Callable
         mean, sigma = get_formatted_stats(pars[i], np.sqrt(cov[i][i]))
         log.info(f"{par_names[i]} = {mean} +/- {sigma}")
     if pad:
-        log.info("")
\ No newline at end of file
+        log.info("")
+
+
+def getenv_bool(name: str, default: bool = False) -> bool:
+    """Get environment value as a boolean, returning True for 1, t and true
+    (caps-insensitive), and False for any other value and default if undefined.
+    """
+    val = os.getenv(name)
+    if not val:
+        return default
+    elif val.lower() in ("1", "t", "true"):
+        return True
+    else:
+        return False
+
+class NumbaMathDefaults(MutableMapping):
+    """Bare-bones class to store some Numba default options. Defaults values
+    are set from environment variables
+
+    Examples
+    --------
+    Set all default option values for a processor at once by expanding the
+    provided dictionary:
+
+    >>> from numba import guvectorize
+    >>> from pygama.math.utils import numba_defaults_kwargs as nb_kwargs
+    >>> @guvectorize([], "", **nb_kwargs, nopython=True) # def proc(...): ...
+
+    Customize one argument but still set defaults for the others:
+
+    >>> from pygama.math.utils import numba_defaults as nb_defaults
+    >>> @guvectorize([], "", **nb_defaults(cache=False) # def proc(...): ...
+
+    Override global options at runtime:
+
+    >>> from pygama.math.utils import numba_defaults
+    >>> # must set options before explicitly importing pygama.math.distributions!
+    >>> numba_defaults.cache = False
+    """
+
+    def __init__(self) -> None:
+        self.parallel: bool = getenv_bool("MATH_PARALLEL", default=True)
+        self.fastmath: bool = getenv_bool("MATH_FAST", default=True)
```

Comment:
Why not moving this module to `pygama.utils`? We could use it for any Numba-decorated routine in pygama. Also, I would replace `MATH_` with `PYGAMA_` here.